cybersyn/scripts/central-planning.lua

```diff
@@ -9,6 +9,36 @@ local band = bit32.band
 local table_remove = table.remove
 local random = math.random
 
+local HASH_STRING = "CS_QUALITY_SEP"
+
+---@param name string The name of the item
+---@param quality string The name of the quality of the item or nil if it is common
+---@return string
+function hash_item(name, quality)
+	if quality == nil then
+		return name
+	else
+		return name .. HASH_STRING .. quality
+	end
+end
+
+---@param sig SignalId
+---@return string
+function hash_signal(sig)
+	return hash_item(sig.name, sig.quality)
```

Comment:
For signals `quality == nil` is the same as `quality == "normal"`. So I think this should be `sig.name, sig.quality or "item"`

Reply:
Adressed in 11a30d6053d5f539bad052dc74ebfc7d8b8360f5

The `hash_item` function handles this.